spec/services/import_polygons_spec.rb

```diff
@@ -1,85 +1,127 @@
+        end
 
-    context "when location type is london boroughs" do
-      let(:location_type) { :london_boroughs }
-      let(:points) { [51.5083356630733, 0.00456249603, 51.5051084208278, -0.0057105307948] }
-      let(:file_name) { "ons_london_boroughs.json" }
+        it "does not import non-existent county" do
+          expect(LocationPolygon.counties.count).to eq(1)
+        end
 
-      it "imports Tower Hamlets local authority" do
-        expect(LocationPolygon.london_boroughs.first.name).to eq("tower hamlets")
+        it "imports Essex boundaries" do
+          expect(LocationPolygon.counties.first.boundary).to eq(boundary_points)
+        end
       end
 
-      it "does not import non-existent local authority" do
-        expect(LocationPolygon.london_boroughs.count).to eq(1)
+      context "when location type is london boroughs" do
+        let(:location_type) { :london_boroughs }
+        let(:boundary_points) { [51.5083356630733, 0.00456249603, 51.5051084208278, -0.0057105307948] }
+        let(:file_name) { "ons_london_boroughs.json" }
+
+        it "imports Tower Hamlets local authority" do
+          expect(LocationPolygon.london_boroughs.first.name).to eq("tower hamlets")
+        end
+
+        it "does not import non-existent local authority" do
+          expect(LocationPolygon.london_boroughs.count).to eq(1)
+        end
+
+        it "imports Tower Hamlets boundaries" do
+          expect(LocationPolygon.london_boroughs.first.boundary).to eq(boundary_points)
+        end
       end
 
-      it "imports Tower Hamlets boundaries" do
-        expect(LocationPolygon.london_boroughs.first.boundary).to eq(points)
+      context "when location type is cities" do
+        let(:location_type) { :cities }
+        let(:boundary_points) { [51.406361958644, -2.3780576677997, 51.4063596372237, -2.3787764623145] }
+        let(:file_name) { "ons_cities.json" }
+
+        it "imports Bath city" do
+          expect(LocationPolygon.cities.first.name).to eq("bath")
+        end
+
+        it "does not import non-existent city" do
+          expect(LocationPolygon.cities.count).to eq(1)
+        end
+
+        it "imports Bath boundaries" do
+          expect(LocationPolygon.cities.first.boundary).to eq(boundary_points)
+        end
       end
     end
 
-    context "when location type is cities" do
-      let(:location_type) { :cities }
-      let(:points) { [51.406361958644, -2.3780576677997, 51.4063596372237, -2.3787764623145] }
-      let(:file_name) { "ons_cities.json" }
-
-      it "imports Bath city" do
-        expect(LocationPolygon.cities.first.name).to eq("bath")
+    context "when not stubbing #get_buffers" do
+      before do
+        distances = [5, 10, 15, 20, 25]
+        distances.each do |distance|
+          allow(HTTParty).to receive(:get).with(
+            "https://ons-inspire.esriuk.com/arcgis/rest/services/Utilities/Geometry/GeometryServer/buffer?"\
```

Comment:
This endpoint is defined elsewhere isn't it?

Reply:
It is, in part, in `import_polygons.rb`, but for the purposes of this test it needs to be different for each distance. How else would I achieve this?